src/server/blockchain/Web3Wallet.js

```diff
@@ -362,18 +361,13 @@ export class Web3Wallet {
         txHash = hash
       }
 
-      let txPromise
-      if (chainId && conf.celoEnabled) {
-        txPromise = this.sendTransaction(this.proxyContract.methods.whitelist(address, did, chainId, 0), {
-          onTransactionHash
-        })
-      } else {
-        txPromise = this.sendTransaction(this.proxyContract.methods.whitelist(address, did), {
-          onTransactionHash
-        })
-      }
+      const txExtraArgs = chainId !== null ? [chainId, 0] : []
+
+      const txPromise = this.sendTransaction(this.proxyContract.methods.whitelist(address, did, ...txExtraArgs), {
+        onTransactionHash
+      })
```

Comment:
you refactoring broke the logic. it is missing conf.celoEnabled
conclusion: STOP REFACTORING
@johnsmith-gooddollar